flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/zookeeper/HighAvailabilityServiceTest.java

```diff
@@ -0,0 +1,115 @@
+import org.apache.flink.runtime.highavailability.HighAvailabilityServices;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobmanager.JobInfo;
+import org.apache.flink.runtime.jobmanager.SubmittedJobGraph;
+import org.apache.flink.runtime.jobmanager.SubmittedJobGraphStore;
+import org.apache.flink.runtime.util.ZooKeeperUtils;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+
+public class HighAvailabilityServiceTest {
+	private TestingServer testingServer;
+	private HighAvailabilityServices zkHaService;
+	private SubmittedJobGraphStore submittedJobGraphStore;
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Before
+	public void before() throws Exception {
+		testingServer = new TestingServer();
+
+		Configuration configuration = new Configuration();
+		configuration.setString(HighAvailabilityOptions.HA_ZOOKEEPER_QUORUM, testingServer.getConnectString());
+		configuration.setString(HighAvailabilityOptions.HA_MODE, "zookeeper");
+		configuration.setString(HighAvailabilityOptions.HA_STORAGE_PATH, "temp");
+		zkHaService = new ZooKeeperHaServices(
+						ZooKeeperUtils.startCuratorFramework(configuration),
+						Executors.directExecutor(),
+						configuration,
+						new FileSystemBlobStore(FileSystem.getLocalFileSystem(), "local"));
+
+		submittedJobGraphStore = zkHaService.getSubmittedJobGraphStore();
+		submittedJobGraphStore.start(new SubmittedJobGraphStore.SubmittedJobGraphListener() {
+			@Override
+			public void onAddedJobGraph(JobID jobId) {
+
+			}
+
+			@Override
+			public void onRemovedJobGraph(JobID jobId) {
+
+			}
+		});
+	}
+
+	@After
+	public void after() throws Exception {
+		testingServer.stop();
+		testingServer = null;
+
+		submittedJobGraphStore.stop();
+		submittedJobGraphStore = null;
+
+		zkHaService.closeAndCleanupAllData();
+		zkHaService = null;
+	}
+
+	/**
+	 * Tests for that the function of cleanupData(JobID) in SubmittedJobGraph
+	 */
+	@Test
+	public void testCleanSubmittedJobGraphStore() throws Exception {
+		SubmittedJobGraph jobGraph1 = new SubmittedJobGraph(
+						new JobGraph("testSubmittedJob1"),
+						new JobInfo(ActorRef.noSender(), ListeningBehaviour.DETACHED, 0, Integer.MAX_VALUE));
+		SubmittedJobGraph jobGraph2 = new SubmittedJobGraph(
+						new JobGraph("testSubmittedJob2"),
+						new JobInfo(ActorRef.noSender(), ListeningBehaviour.DETACHED, 0, Integer.MAX_VALUE));
+		submittedJobGraphStore.putJobGraph(jobGraph1);
+		submittedJobGraphStore.putJobGraph(jobGraph2);
+
+		zkHaService.cleanupData(jobGraph1.getJobId());
+
+		SubmittedJobGraph recoverJobGraph2 = submittedJobGraphStore.recoverJobGraph(jobGraph2.getJobId());
+		Assert.assertEquals(recoverJobGraph2.getJobId(), jobGraph2.getJobId());
+		thrown.expectMessage("Could not retrieve the submitted job graph state handle for /" +
```

Comment:
Could we rather check for the exception type? Matching exception messages is really brittle.